### ISSUE TITLE
Proxy using fetch under the hood?

### ISSUE THREAD

**Issue by tom-drake** [jameslnewell/xhr-mock#78]
I noticed that if a request is proxied it is done as a fetch rather than an XHR. Would it make sense just to use the RealXMLHttpRequest rather than 'http' and 'https'?

**Comment by jameslnewell** [comment]
Yep sounds reasonable. From memory, because we've replaced the global XMLHttpRequest, we don't have access to the real XMLHttpRequest within a handler so using `fetch` was a hack to send real requests instead of mock ones.

In v3 the handler will receive a context object that we can put the real XMLHttpRequest on.

**Comment by tom-drake** [comment]
If this is going into a future version then great. I worked around this by creating my own proxy that makes use of XHRMock.RealXMLHttpRequest which seems to give me access to the real one before being overridden.
